GetIntoTeachingApi/Utils/LoggableMessageComposer.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using GetIntoTeachingApi.Attributes;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace GetIntoTeachingApi.Utils
+{
+    public class LoggableMessageComposer
+    {
+        public static string LogMessage(string prefix, ControllerActionDescriptor descriptor, IEnumerable<string> messages)
+        {
+            var actionName = descriptor.ActionName;
+            var controllerName = descriptor.ControllerName;
+            var payload = string.Join("\n", messages.Where(m => !string.IsNullOrEmpty(m)));
+
+            return $"{prefix} {controllerName}:{actionName} - {payload}";
+        }
+
+        public static string LogMessageForObject(object obj)
+        {
+            var type = obj.GetType().GetGenericArguments().FirstOrDefault();
+
+            if (type == null)
```

Comment:
If this is set to `.FirstOrDefault()` just above will it ever be `null`?

Reply:
Yeah, so if we did `.First()` here it would throw an exception if there were no elements, but `FirstOrDefault()` really means first or `null` if there are no elements. Not very clear naming by LINQ IMO...

Reply:
Ah, makes sense. Thanks 👍🏽